src/main/java/Duke.java

```diff
@@ -1,10 +1,226 @@
+                    break;
+                }
+            } catch (DukeException error) {
+                System.out.println(error);
+            }
+            printLine();
+            command = new Command(in.nextLine());
+        }
+
+    }
+
+    /** Print the welcome massage.
+     *
+     */
+    private static void printWelcomeMessage() {
+
+        String logo = " __A__   _C__   _E__\n"
+                    + "/  _  \\ /  __| |  __|\n"
+                    + "| |_| | | |    | |__\n"
+                    + "|  _  | | |__  | |__\n"
+                    + "| | | | \\____| |____|\n";
+
+        System.out.print(logo);
+        printLine();
+
+        System.out.println("\tHey there! I'm Ace, your very own personal assistant.");
+        System.out.println("\tHow can I help you today?\n");
+        printLine();
+
+    }
+
+    /** Adds the new task to the list.
+     *
+     * @param command task name given by the user
+     * @param tasks the list of tasks
+     * @param type the type of task
+     */
+    private static void executeAdd(String command, Task[] tasks, CommandType type) throws DukeException {
+
+        switch (type) {
+        case TODO:
+            try {
+                String todoName = command.substring(5).trim();
+                tasks[Task.getNoOfTasks()] = new Todo(todoName);
+            } catch (StringIndexOutOfBoundsException e) {
+                throw new DukeException(ExceptionType.MISSING_DESCRIPTION);
+            }
+            break;
+        case DEADLINE:
+            String[] rawName = command.trim().split(DEADLINE_IDENTIFIER);
+            if (rawName[0].trim().equalsIgnoreCase("deadline")) {
+                throw new DukeException(ExceptionType.MISSING_DESCRIPTION);
+            }
+            if (!command.contains(DEADLINE_IDENTIFIER)) {
+                throw new DukeException(ExceptionType.MISSING_IDENTIFIER);
+            }
+            if(rawName.length<2) {
+                throw new DukeException(ExceptionType.MISSING_ON_BY);
+            }
+            String deadlineName = rawName[0].substring(9).trim();
+            String by = rawName[1].trim();
+            tasks[Task.getNoOfTasks()] = new Deadline(deadlineName, by);
+            break;
+        case EVENT:
+            String[] rawEventName = command.trim().split(EVENT_IDENTIFIER);
+            if (rawEventName[0].trim().equalsIgnoreCase("event")) {
+                throw new DukeException(ExceptionType.MISSING_DESCRIPTION);
+            }
+            if (!command.contains(EVENT_IDENTIFIER)) {
+                throw new DukeException(ExceptionType.MISSING_IDENTIFIER);
+            }
+            if(rawEventName.length<2) {
+                throw new DukeException(ExceptionType.MISSING_ON_BY);
+            }
+            String eventName = rawEventName[0].substring(6).trim();
+            String on = rawEventName[1].trim();
+            tasks[Task.getNoOfTasks()] = new Event(eventName, on);
+            break;
+        }
+    }
```

Comment:
Maybe you can try extracting out certain lines of code to shorten the method? It seems a bit long

Reply:
Thanks for pointing out!